test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+// !\author: Clemens Cords <clemens.cords@fu-berlin.de>
+
+#include <benchmark/benchmark.h>
+#include <cstring>
+#include <fstream>
+
+#include <seqan3/test/seqan2.hpp>
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/seq_io.h>
+#endif
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/io/sequence_file/format_fastq.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/std/filesystem>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+using namespace seqan3;
+
+constexpr unsigned seed = 1234u;
+
+// ============================================================================
+// generate fastq file
+// ============================================================================
+constexpr size_t default_sequence_length = 50; //length of nucleotide and quality sequence
+
+std::string get_file(size_t n_entries)
+{
+    std::string file{};
+    std::string const id{"@name"};
+
+    for (size_t i = 0; i < n_entries; ++i)
+    {
+        auto seq = test::generate_sequence<dna5>(default_sequence_length, 0, seed);
+        std::string seq_string = seq | view::to_char;
```

Comment:
Yes but we want that?! so our benchmarks are comparable across builds?

Reply:
If you have a fixed seed at the start of the function and change the seed in a deterministic fashion the whole process will still be deterministic.

```c++
auto seq = test::generate_sequence<dna5>(default_sequence_length, 0, i); // this is deterministic
// but every sequence will be different
```

Reply:
I thought that for each call of ``generate_sequence`` function the MT rng engine would advance it's state but since it's constructed newly for each call all the sequences are literally the same. I fixed it now